Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1,84 +1,64 @@
 
-#### Static Analysis
+- [Ghidra](https://ghidra-sre.org/ "Ghidra") is a software reverse engineering (SRE) suite of tools developed by NSA's Research Directorate. Please refer to the [installation guide](https://ghidra-sre.org/InstallationGuide.html "Ghidra Installation Guide") on how to install it and look at the [cheat sheet](https://ghidra-sre.org/CheatSheet.html "Cheat Sheet") for a first overview of available commands and shortcuts.
 
-##### Recovering an IPA File From an Installed App
+##### Building a Reverse Engineering Environment for Free
 
-###### From Jailbroken Devices
-
-You can use Saurik's [IPA Installer Console](https://cydia.saurik.com/package/com.autopear.installipa/ "IPA Installer Console") to recover IPAs from apps installed on the device. To do this, install `IPA Installer Console` via Cydia. Then, SSH into the device and look up the bundle ID of the target app. For example through listing of the available apps:
+Be sure to follow the instructions from the section "Setting up Xcode and Command Line Tools" of chapter "iOS Basic Security Testing". This way you'll have properly installed [Xcode](https://developer.apple.com/xcode/ide/ "Apple Xcode IDE"). We'll be using standard tools that come with macOS and Xcode in addition to the tools mentioned above. Make sure you have the [Xcode command line developer tools](https://railsapps.github.io/xcode-command-line-tools.html "Xcode Command Line Tools") properly installed or install them straight away from your terminal:
 
 ```shell
-iPhone:~ root# ipainstaller -l
-com.apple.Pages
-com.example.targetapp
-com.google.ios.youtube
-com.spotify.client
+$ xcode-select --install
 ```
 
-Generate the IPA file via the following command:
+- [`xcrun`](http://www.manpagez.com/man/1/xcrun/ "xcrun man page") can be used invoke Xcode developer tools from the command-line, without having them in the path. For example you may want to use it to locate and run swift-demangle or simctl.
+- swift-demangle is an Xcode tool that demangles Swift symbols. For more information run `xcrun swift-demangle -help` once installed.
+- simctl is an Xcode tool that allows you to interact with iOS simulators via the command line to e.g. manage simulators, launch apps, take screenshots or collect their logs.
 
-```shell
-iPhone:~ root# ipainstaller -b com.example.targetapp -o /tmp/example.ipa
-```
+##### Commercial Tools
 
-###### From Non-Jailbroken Devices
+Although working with a completely free setup is possible, you should consider investing in commercial tools. The main advantage of these tools is convenience: they come with a nice GUI, lots of automation, and end user support. If you earn your daily bread as a reverse engineer, they will save you a lot of time.
```

Comment:
The first sentence here is a bit odd: can you rephrase this paragraph, because the overall idea of the paragraph (doing reverse engineering? then a commercial tool can help) is nice, but i am sure you don't need it if you already have enough support with the free tools available...

Reply:
I agree, this is the actual version in master. I've changed it in both 5c and 6c to:

```
Building a reverse engineering environment for free is possible. However, there are some commercial alternatives. The most commonly used are:
```

what do you think?